cmd/adapter/adapter.go

```diff
@@ -119,7 +119,13 @@ func (cmd *PrometheusAdapter) makePromClient() (prom.Client, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to read prometheus-token-file: %v", err)
 		}
-		httpClient.Transport = transport.NewBearerAuthRoundTripper(string(data), httpClient.Transport)
+		var wrappedTransport http.RoundTripper
+		if httpClient.Transport != nil {
+			wrappedTransport = httpClient.Transport
+		} else {
+			wrappedTransport = http.DefaultTransport
+		}
```

Comment:
nit: small suggestion
```suggestion
		wrappedTransport := http.DefaultTransport
		if httpClient.Transport != nil {
			wrappedTransport = httpClient.Transport
		}
```